rr_computer_vision/src/lane_detection.cpp

```diff
@@ -20,4 +33,144 @@ LaneDetection::LaneDetection() {
 */
 LaneDetection::~LaneDetection() {
 
+}
+
+void LaneDetection::InitializeSubscribers() {
+    zed_subscriber = nh_.subscribe("/zed/rgb/image_rect_color", 1, &LaneDetection::RGBCameraCallback, this);
+}
+
+void LaneDetection::InitializePublishers() {
+    test_publisher = nh_.advertise<sensor_msgs::Image>("/test_publisher", 1);
+    pointList_pub_ = nh_.advertise<nav_msgs::OccupancyGrid>("/point_vec_out_", 1);
+}
+
+void LaneDetection::RGBCameraCallback(const sensor_msgs::Image& msg){
```

Comment:
Also I did not mention this before and I take partial responsibility for it but do not make callbacks this long. Our callbacks should be very short (only copying data) and these other stuff should not be occuring in the callback.